lib/sda/node.go

```diff
@@ -125,6 +127,50 @@ func (n *Node) SendTo(to *TreeNode, msg interface{}) error {
 	return n.overlay.SendToTreeNode(n.token, to, msg)
 }
 
+// SendToParent sends a given message to the parent of the calling node (unless it is the root)
+func (n *Node) SendToParent(msg interface{}) error {
+	if !n.IsRoot() {
+		if err := n.SendTo(n.Parent(), msg); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
Just a stylistic remark: We should prefer doing the *invalid* path first inside the `if` and the rest in the normal track of the function:
```
func (n *Node) SendToParent(msg interface{}) error {
    if n.IsRoot() {
        return nil
    }
    return n.SendTo(n.Parent(),msg)
}
```
I'd say the same for the rest of the functions .